### ISSUE TITLE
Add more texture format support

### ISSUE THREAD

**Issue by Redwarx008** [godotengine/godot-proposals#7540]
### Describe the project you are working on

A large terrain using heightmap.

### Describe the problem or limitation you are having in your project

The height map used by the terrain is usually stored as a 16-bit unsigned int, but godot only supports R16_sfloat. If you want to not lose precision, you can only use R32_sfloat, which will double the size.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add uint support for all uncompressed formats, they can be sampled, just linear filtering is not supported. But I guess, people who use them should know this.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

-

### If this enhancement will not be used often, can it be worked around with a few lines of script?

can not

### Is there a reason why this should be core and not an add-on in the asset library?

it's render core

**Comment by clayjohn** [comment]
Is this proposal specifically for the Image class? Its a bit unclear where you are asking for the support to be implemented and what it will be used for

**Comment by Redwarx008** [comment]
> Is this proposal specifically for the Image class? Its a bit unclear where you are asking for the support to be implemented and what it will be used for

 Currently, the main way for Godot to use textures is through image classes, so adding them to image classes is the first step, followed by modifying the format check in different places .

**Comment by TokisanGames** [comment]
> Is this proposal specifically for the Image class? 

Yes, this proposal should be renamed and understood to be: Support all standard opengl integer and float formats in `Image` and `Texture` classes. 8/16/32 uint/sint at the least. The vulkan rendering device already supports them. OpenGL should as well.

As a workaround, we can currently write integers into FORMAT_RF and read them in the shader as ints with this process:
https://github.com/godotengine/godot/issues/57841#issuecomment-1773930802

> It's a bit unclear where you are asking for the support to be implemented and what it will be used for

What it is used for is integer data like bit fields. In Terrain3D, I am currently [packing tons of information into 32 bits](https://github.com/TokisanGames/Terrain3D/issues/115) with the above workaround.
